File: basics/generate-text/typescript/example-multi.ts
#!/usr/bin/env -S npx ts-node --transpileOnly
import { Substrate, MultiComputeText } from "substrate";

async function main() {
  const apiKey = process.env["SUBSTRATE_API_KEY"] || "YOUR_API_KEY";
  const substrate = new Substrate({ apiKey: apiKey });

  const story = new MultiComputeText({
    prompt: "tell me a short 2-sentence story",
    num_choices: 2,
  });

  const start = logStart();
  const res = await substrate.run(story);
  logEnd(start);

  console.log("1:", res.get(story).choices[0].text);
  console.log("2:", res.get(story).choices[1].text);
}

main();

function logStart(): number {
  const start = Date.now();
  console.log("\x1b[94m֍ \x1b[34m Running... \x1b[0m");
  return start;
}

function logEnd(start: number) {
  const end = Date.now();
  const duration = end - start;
  const seconds = (duration / 1000).toFixed(2);
  console.log(`\x1b[92m֍ \x1b[32m Completed in ${seconds}s\x1b[0m`);
}

File: basics/generate-text/python/src/generatetext/example_multi.py
import os
import time
from substrate import Substrate, MultiComputeText


def main():
    api_key = os.environ.get("SUBSTRATE_API_KEY") or "YOUR_API_KEY"
    substrate = Substrate(api_key=api_key)

    story = MultiComputeText(
        prompt="tell me a short 2-sentence story",
        num_choices=2,
    )

    start = log_start()
    res = substrate.run(story)
    log_end(start)

    print("1:", res.get(story).choices[0].text)
    print("2:", res.get(story).choices[1].text)


def log_start():
    start = time.time()
    print("\033[94m֍ \033[34mRunning...\033[0m")
    return start


def log_end(start):
    end = time.time()
    duration = end - start
    seconds = f"{duration:.2f}"
    print(f"\033[92m֍ \033[32mCompleted in {seconds}s\033[0m")


if __name__ == "__main__":
    main()
